lib/beacon/live_admin/live/page_editor_live/revisions.ex

```diff
@@ -26,70 +45,114 @@ defmodule Beacon.LiveAdmin.PageEditorLive.Revisions do
-              <div class="w-full mt-2">
-                <div class="py-3 bg-[#282c34] rounded-lg">
-                  <LiveMonacoEditor.code_editor
-                    path={event.snapshot.id <> "-schema"}
-                    style="min-height: 200px; width: 100%;"
-                    value={Jason.encode!(event.snapshot.page.raw_schema, pretty: true)}
-                    opts={Map.merge(LiveMonacoEditor.default_opts(), %{"language" => "json", "readOnly" => "true"})}
-                  />
-                </div>
-              </div>
-            </li>
-            <li>
-              <h4 class="text-gray-600">Meta Tags</h4>
-              <%= render_meta_tags(event.snapshot.page.meta_tags) %>
-            </li>
-          </ol>
-        </li>
+        <%= for event <- @events do %>
+          <.revision event={event} />
+        <% end %>
       </ol>
+
+      <.modal :if={@show_variant_modal} id="variant-modal" on_cancel={JS.push("hide_modal")} show>
+        <div class="w-full mt-2">
+          <div class="py-3 bg-[#282c34] rounded-lg">
+            <LiveMonacoEditor.code_editor
+              path="variant_template"
+              style="min-height: 200px; width: 100%;"
+              value={@variant_template}
+              opts={Map.merge(LiveMonacoEditor.default_opts(), %{"language" => "json", "readOnly" => "true"})}
+            />
+          </div>
+        </div>
+      </.modal>
     </div>
     """
   end
 
+  ## FUNCTION COMPONENTS
+
+  attr :event, :map
+  attr :latest, :boolean, default: false
+
+  def revision(assigns) do
+    ~H"""
+    <li class="group mb-10 ml-6">
+      <span class="absolute flex items-center justify-center w-6 h-6 bg-blue-100 rounded-full -left-3 ring-8 ring-white">
+        <.icon :if={@event.event == :published} name="hero-eye-solid" class="h-4 w-4 text-blue-800" />
+        <.icon :if={@event.event == :created} name="hero-document-plus-solid" class="h-4 w-4 text-blue-800" />
+      </span>
+      <h3 class="flex items-center mb-1 text-lg font-semibold text-gray-900">
+        <%= Phoenix.Naming.humanize(@event.event) %> <span class="text-sm text-gray-500 ml-2"><%= format_datetime(@event.inserted_at) %></span>
+        <span class="hidden group-first:block bg-blue-100 text-blue-800 text-sm font-medium mr-2 px-2.5 py-0.5 rounded ml-3">Latest</span>
+      </h3>
+
+      <ol :if={@event.snapshot} class="space-y-3">
+        <li>
+          <h4 class="text-gray-600 text-bold">Path</h4>
+          <%= @event.snapshot.page.path %>
+        </li>
+        <li>
+          <h4 class="text-gray-600">Title</h4>
+          <%= @event.snapshot.page.title %>
+        </li>
+        <li>
+          <h4 class="text-gray-600">Description</h4>
+          <%= @event.snapshot.page.description %>
+        </li>
+        <li>
+          <h4 class="text-gray-600">Format</h4>
+          <%= @event.snapshot.page.format %>
+        </li>
+        <li>
+          <h4 class="text-gray-600">Body</h4>
```

Comment:
I think I made a typo in the original template, this element should be:

```suggestion
          <h4 class="text-gray-600">Template</h4>
```